src/server/job_func.c

```diff
@@ -1992,295 +1954,6 @@ job_or_resv_init_wattr(void *pobj, int obj_type)
-
-	(void)strcpy(presv->ri_qs.ri_fileprefix,
-		pjob->ji_qs.ji_jobid);
-
-
-	/* A resc_resv's "ri_queue" is not the empty string if a
-	 * queue has been specifically established to support the
-	 * reservation's job(s)
-	 */
-	(void)strcpy(presv->ri_qs.ri_queue, "");
-	presv->ri_qp = NULL;
-
-	/* Now set various of the reservation attributes
-	 * based on what is specified for the job
-	 * Makes use of the correspondence table located
-	 * in resv_attr_def.c
-	 */
-	rc =  set_resvAttrs_off_jobAttrs(presv, pjob);
-	if (rc != 0) {
-		resv_purge(presv);
-		return (PBSE_INTERNAL);
-	}
-
-	/* Set any other needed attributes and resc_resv fields */
-
-
-	/* set reservation "owner" attribute to user@host */
-	resv_attr_def[(int)RESV_ATR_resv_owner].at_free(
-		&presv->ri_wattr[(int)RESV_ATR_resv_owner]);
-	(void)strcpy(buf, pjob->ji_wattr[(int)JOB_ATR_job_owner].at_val.at_str);
-	resv_attr_def[(int)RESV_ATR_resv_owner].at_decode(
-		&presv->ri_wattr[(int)RESV_ATR_resv_owner],
-		NULL, NULL, buf);
-
-	/* make sure owner is put in reservation's User_List */
-	if (act_resv_add_owner(&presv->ri_wattr[(int)RESV_ATR_userlst],
-		presv, ATR_ACTION_NEW)) {
-		resv_purge(presv);
-		return (PBSE_BADATVAL);
-	}
-
-
-	presv->ri_wattr[(int)RESV_ATR_resv_type]
-	.at_val.at_long = presv->ri_qs.ri_type;
-	presv->ri_wattr[(int)RESV_ATR_resv_type].at_flags |= ATR_VFLAG_SET |
-		ATR_VFLAG_MODIFY | ATR_VFLAG_MODCACHE;
-
-	eval_resvState(presv, RESVSTATE_add_resc_resv_to_job,
-		0, &state, &sub);
-	(void)resv_setResvState(presv, state, sub);
-
-	/* process reservation window, duration, wall info */
-
-	if (start_end_dur_wall(presv->ri_jbp, RESV_JOB_OBJECT)) {
-		(void)resv_purge(presv);
-		return (PBSE_SYSTEM);
-	}
-
-	/*For any resource limits that aren't specified,
-	 *see if a default value can be determined from
-	 *queue, server, etc. and use that value for the limit
-	 */
-
-	(void)set_resc_deflt((void *)presv, RESC_RESV_OBJECT, NULL);
-
-	/* write of disk image for this resc_resv structure
-	 * occurs when function job_or_resv_save () is called
-	 */
-
-	/* put onto the "timed task" list a task that causes
-	 * deletion of the reservation if the window passes
-	 */
-
-	if (gen_task_EndResvWindow(presv)) {
-		(void)resv_purge(presv);
-		return (PBSE_SYSTEM);
-	}
-
-	/*
-	 *We don't want to attach the reservation to the job at this
-	 *point but rather in "req_commit" where the job is taken from
-	 *the servers's "svr_newjobs" list and placed on the server's
-	 *"svr_alljobs" list.
-	 *This avoids a timing issue where "stat" requests coming from
-	 *the scheduler and the "req_commit" request can be such that
-	 *the scheduler sees the reservation before the reservation's
-	 *job is on svr_alljobs.  Immediate deleting of such a reservation
-	 *ends up deleting the reservation but leaves the job structure
-	 *still in existence (on the svr_alljobs list) following the
-	 *req_commit request).
-	 */
-
-	append_link(&svr_newresvs, &presv->ri_allresvs, presv);
```

Comment:
Please remove `svr_newresvs` global list, as it was only used for "reservation job" only

Reply:
Thanks for catching that!